### ISSUE TITLE
CoreV1Api.read_node() crashes upon encountering a <none> image

### ISSUE THREAD

**Issue by orencohendev** [kubernetes-client/python#1422]
**What happened (please include outputs or screenshots)**:
CoreV1Api.read_node() excepts and fails when an image is `<none>`.
Traceback shown below.
The gist of the bug is that the method tries to access a property of the image and it crashes because it's None.

![Screenshot from 2021-04-12 13-08-45](https://user-images.githubusercontent.com/62210374/115670701-614cc900-a352-11eb-8bbd-9e24df10fb40.png)

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/tornado/web.py", line 1702, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/src/app/handlers/status_handler.py", line 9, in get
    stats = SystemManager.get_system_stats()
  File "/src/app/services/system_manager.py", line 60, in get_system_stats
    node_stats = SystemManager.get_node_stats(node)
  File "/src/app/services/system_manager.py", line 69, in get_node_stats
    read_node = core_api.read_node(node["metadata"]["name"])
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api/core_v1_api.py", line 24553, in read_node
    return self.read_node_with_http_info(name, **kwargs)  # noqa: E501
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api/core_v1_api.py", line 24640, in read_node_with_http_info
    return self.api_client.call_api(
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 348, in call_api
    return self.__call_api(resource_path, method,
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 192, in __call_api
    return_data = self.deserialize(response_data, response_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 264, in deserialize
    return self.__deserialize(data, response_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 303, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 639, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 303, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 639, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 280, in __deserialize
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 280, in <listcomp>
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 303, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 641, in __deserialize_model
    instance = klass(**kwargs)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/models/v1_container_image.py", line 55, in __init__
    self.names = names
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/models/v1_container_image.py", line 80, in names
    raise ValueError("Invalid value for `names`, must not be `None`")  # noqa: E501
ValueError: Invalid value for `names`, must not be `None`

```

**What you expected to happen**:
The function should be able to recover if an image is None.

**How to reproduce it (as minimally and precisely as possible)**:
I don't know how to reproduce a `<none>` image but the issue can be reproduced when you have a `<none>` image for whatever reason appearing in `k3s crictl images` and you try to access the method like so:

```
CoreV1Api.read_node("Node Name")
```

**Anything else we need to know?**:
I'll add more info here as you request it.

**Environment**:
- Kubernetes version: 1.18
- OS: Ubuntu 18.04.5
- Python version: Python 3.9.1
- Python client version: 12.0.1

**Comment by roycaihw** [comment]
Please let me know when you have a reproduce of the except message

**Comment by orencohendev** [comment]
Updated the bug description with the traceback I got.

**Comment by jony89** [comment]
+1

**Comment by roycaihw** [comment]
Thanks for the information. This seems to be a duplicate of https://github.com/kubernetes/kubernetes/issues/100802. The issue has been fixed upstream in k/k 1.22 https://github.com/kubernetes/kubernetes/pull/102159. In this Python client we are still working on catching up with the k/k releases. We will be releasing 1.20 clients soon. The issue will be fixed when we generate 1.22 clients.
